### ISSUE TITLE
Issue with bap 0.9.9 - llvm::TargetRegistry::lookupTarget

### ISSUE THREAD

**Issue by abarbaro** [BinaryAnalysisPlatform/bap#545]
Hi,
I'm sorry to post a similar issue but I cannot find the proper solution.

Once I try to install bap from the repository I have the following error:
```
lib/bap_disasm/libllvm_stubs.a(llvm_disasm.o): In function `bap::llvm_disassembler::create(char const*, char const*, int)':
/home/username/.opam/4.02.1/build/bap.0.9.9/_build/lib/bap_disasm/llvm_disasm.cpp:123: undefined reference to `llvm::TargetRegistry::lookupTarget(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'
```
As you suggested last time I've specified the version of LLVM that I wanted to use with the following command

```
llvm_config=/usr/bin/llvm-config-3.4 opam install bap
```

Am I missing something again?

Thanks

**Comment by ivg** [comment]
It looks like a linker error... Can you provide the following information:
1. OS/distribution, that you're using
2. Build and configuration logs, they should be stored by opam in $opam/$switch/build/bap folder as far as I remember.

**Comment by abarbaro** [comment]
Hi,
yes I'm using Ubuntu 16.04.

I have only the setup.data
[setup.data.zip](https://github.com/BinaryAnalysisPlatform/bap/files/439404/setup.data.zip)

Thanks for your help

**Comment by ivg** [comment]
How did you install llvm-3.4 on Ubuntu 16.04?

**Comment by abarbaro** [comment]
Hi,
I have just used apt. Before I added the repositories following these instructions: http://askubuntu.com/questions/405765/installing-clangllvm-3-4-from-llvm-org-into-saucy-13-10

Obv I used the repo for my version of Ubuntu.

Thanks

**Comment by ivg** [comment]
Yeah, it looks like that the problem lies outside of the bap and our build system. You configuration is fine, and the problem is at the linker stage, indicating that the llvm installation is broken. 

The good news is that we were working past two months on llvm-3.8 support. Currently it builds without a problem on Ubuntu Xenial. Right now I'm packaging it into opam. (It takes some time, I've already installed bap today few dozens times). 

So, I think that it is better to wait until we will have a native support for Xenial. Especially, since 0.9.9 is very outdated.

**Comment by abarbaro** [comment]
OK, when would it be ready?

I'll try to reinstall llvm from the source code. Can I after that use the latest version of bap with it?

Thanks

**Comment by ivg** [comment]
there're slight chances that it will be ready today, otherwise on Monday.

**Comment by ivg** [comment]
Ok, it is done. Here is the sequence of commands, that is needed to install bap on freshly cloned xenial box:
```
    $ history
    1  sudo apt-get install opam
    2  opam init 
    3  eval `opam config env`
    4  opam repo add bap git://github.com/BinaryAnalysisPlatform/opam-repository.git
    5  opam depext bap
    6  opam install bap
```

**Comment by abarbaro** [comment]
Hi,
I tried on another VM, just created a couple of days ago and I have the same error. Probably I'm missing something.

Thanks

**Comment by ivg** [comment]
Are you talking about 1.0.0 version? If so, then I suggest to get rid of manually installed llvm or, even better, stay from a fresh new VM

**Comment by abarbaro** [comment]
Hi,
no I tried bap 0.9.9. I'll prepare a new VM and fix the problem.

Thanks

**Comment by ivg** [comment]
fixed in #546.
